File: bukkit/src/main/java/io/wdsj/asw/bukkit/util/PlayerUtils.java
package io.wdsj.asw.bukkit.util;

import org.bukkit.entity.Player;

import static io.wdsj.asw.bukkit.util.Utils.isAnyClassLoaded;

public class PlayerUtils {
    private static final boolean isLeavesServer = isAnyClassLoaded("top.leavesmc.leaves.LeavesConfig", "org.leavesmc.leaves.LeavesConfig");

    public static boolean isNpc(Player player) {
        if (isLeavesServer) {
            return player.getAddress() == null || player.hasMetadata("NPC");
        } else {
            return player.hasMetadata("NPC");
        }
    }

    private PlayerUtils() {}
}

File: bukkit/src/main/kotlin/io/wdsj/asw/bukkit/util/PlayerUtils.kt
package io.wdsj.asw.bukkit.util

import org.bukkit.entity.Player

object PlayerUtils {
    private val isLeavesServer =
        Utils.isAnyClassLoaded("top.leavesmc.leaves.LeavesConfig", "org.leavesmc.leaves.LeavesConfig")

    fun isNpc(player: Player): Boolean {
        return if (isLeavesServer) {
            player.address == null || player.hasMetadata("NPC")
        } else {
            player.hasMetadata("NPC")
        }
    }
}
